python/hail/typecheck.py

```diff
@@ -0,0 +1,123 @@
+from decorator import decorator, getargspec
+from types import ClassType
+import re
```

Comment:
This should be in its own package defined in __init__ and declare exports so people who import * from typecheck don't get extract and DictChecker and other things.